src/JsCube.js

```diff
@@ -72,10 +86,14 @@ class JsCube {
     return this.result.dimensions.map(dimDict => new Dimension(dimDict));
   }
 
-  get dimensionTypes() {
+  get allDimensionTypes() {
     return this.dimensions.map(dim => dim.type);
   }
 
+  get dimensionTypes() {
```

Comment:
Please rename this file with `JsCube` class definition into `cube.js`, with making sure that all tests pass. We'll then move those "vector" classes into separate files in one of the following PRs.